lib/harvester.js

```diff
@@ -310,6 +309,40 @@ Harvester.prototype.after = function (name, fn) {
   return this;
 };
 
+Harvester.prototype.exportRoles = function () {
+    var roles = {};
+    _.forEach(this.createdResources, function (resource) {
+        _.forEach(resource.roles, function (role) {
+            roles[role] = roles[role] || [];
+            route.getRestMethods().forEach(function (httpMethod) {
+                if (resource[httpMethod]().options.notAllowed) {
+                        return;
+                    }
+                    var methodRoles = resource[httpMethod]().roles;
+                    var permission = resource[httpMethod]().getPermission();
+                    if (_.isEmpty(methodRoles)) {
+                        roles[role].push(permission);
+                    } else {
+                        _.forEach(methodRoles, function (role) {
+                            roles[role] = roles[role] || [];
```

Comment:
These two blocks require deep thought to figure out what they're doing - lets add some nice comments about the intention of the code in here.